Integracion formulario materia

gregorionunez8 opened the pull request.

gregorionunez8 committed:
Correccion de MateriaData e integracion de FormularioMateria

gregorionunez8 committed:
Update MateriaData.java

gregorionunez8 committed:
Merge branch 'main' into Integracion-FormularioMateria

gregorionunez8 committed:
Coreccion de Menu.java e integracion de FormularioMateria

MaribelAngulo commented on UniversidadGrupo67/src/AccesoADatos/MateriaData.java:
```diff
@@ -73,51 +73,56 @@ public Materia buscarMateria(int id) {
-    public void eliminarMateria(int id) {
-        if(materiaExiste(id)){
-        // Consulta sql para insertar materia
+    public void eliminarMateria(Materia materia) {
```
el parametro que debe recibir es el id de la materia y debe ser tipo int

gregorionunez8 replied:
Envio la materia entera para corroborar si ya esta eliminada o no, y poder decir si ya esta eliminada o no. Si envio solo el id tendria que hacer una consulta SELECT para corroborarlo.

MaribelAngulo commented on UniversidadGrupo67/src/AccesoADatos/MateriaData.java:
```diff
@@ -73,51 +73,56 @@ public Materia buscarMateria(int id) {
+            ps.setInt(1, id); //Asignacion de valores
             ResultSet rs = ps.executeQuery();
-            while (rs.next()) {
+            if (rs.next()) { //Verificar si existe una materia con ese codigo
```
este cambio es correcto?

gregorionunez8 replied:
Si es correcto, porque es solo una fila . De las dos maneras es correcto.

MaribelAngulo commented on UniversidadGrupo67/src/AccesoADatos/MateriaData.java:
```diff
@@ -73,51 +73,56 @@ public Materia buscarMateria(int id) {
+            } else {
+                JOptionPane.showMessageDialog(null, "No existe una materia con ese codigo"); //Dialogo si no existe una materia con ese codigo
             }
-            ps.close();
```
si esta linea queda eliminada donde se cierra ps?

gregorionunez8 replied:
Me olvide de cerrarlo. Ahora lo corrijo

MaribelAngulo commented on UniversidadGrupo67/src/AccesoADatos/MateriaData.java:
```diff
@@ -73,51 +73,56 @@ public Materia buscarMateria(int id) {
-    // Metodo para listar materias
-    public List listarMaterias() {
+// Metodo para listar materias
+public List listarMaterias() {
```
`List<Materia>`

gregorionunez8 replied:
No es necesario asignarle el tipo de objeto en el metodo

gregorionunez8 committed:
Agregue y correji

Agregue los ps.close()
Agregue que te devuelva el codigo de la materia agregada
Borre codigo innecesario
